src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1979,8 +1979,9 @@ open_common(int (*real_open)(const char *pathname, int oflags, ...), const char
 	}
 	/* file/dir should be handled by DFS */
 	if (oflags & O_CREAT) {
-		rc = dfs_open(dfs_mt->dfs, parent, item_name, mode | S_IFREG, oflags & (~O_APPEND),
-			      0, 0, NULL, &dfs_obj);
+		/* clear the bits for types first */
+		rc = dfs_open(dfs_mt->dfs, parent, item_name, (mode & (~S_IFMT)) | S_IFREG,
```

Comment:
another question, what if someone really specifies an invalid mode_t and we need to return an error?

Reply:
mode is not checked here. dfs_open needs to check the mode.

Reply:
There is no type bit in mode in glibc open(). Only permission bit there.
dfs_open() combines permission and type bits.